src/structures/APIMessage.js

```diff
@@ -159,12 +186,45 @@ class APIMessage {
       }
     }
 
-    return Promise.all(fileLikes.map(f => this.constructor.resolveFile(f)));
+    this.files = await Promise.all(fileLikes.map(f => this.constructor.resolveFile(f)));
+    return this;
+  }
+
+  /**
+   * Converts this APIMessage into an array of APIMessages for each split content
+   * @returns {APIMessage[]}
+   */
+  split() {
+    if (!this.data) this.resolveData();
+
+    if (!(this.data.content instanceof Array)) return [this];
+
+    const apiMessages = [];
+
+    for (let i = 0; i < this.data.length; i++) {
+      let data;
+      let opt;
+
+      if (i === this.data.content.length - 1) {
+        data = Object.assign({}, this.data, { content: this.data.content[i] });
+        opt = this.options;
+      } else {
+        data = Object.assign({}, this.data, { content: this.data.content[i], embed: undefined, embeds: undefined });
+        opt = Object.assign({}, this.options, { files: undefined });
+      }
```

Comment:
It doesn't matter in the end, but you should ensure this section is correct for either `MessageOptions` or `WebhookMessageOptions` in my opinion.